app/src/main/java/kth/jjve/xfran/SaveResultsActivity.java

```diff
@@ -40,53 +55,103 @@ protected void onCreate(Bundle savedInstanceState) {
         mExercises = findViewById(R.id.workout_exercises);
-
-        saveButton = findViewById(R.id.save_button);
-        cancelButton = findViewById(R.id.cancel_save_button);
         workoutItemSaveButton = findViewById(R.id.button_save_wod);
         workoutItemPlanButton = findViewById(R.id.button_plan_wod);
+        //save results
+        mDate = findViewById(R.id.edit_date);
+        mScore = findViewById(R.id.edit_score);
+        mComments = findViewById(R.id.edit_comments);
+        mScoreType = findViewById(R.id.type_text);
+        mScaledSwitch = findViewById(R.id.scaled_switch);
+        mScoreSeekBar = findViewById(R.id.score_bar);
+        saveButton = findViewById(R.id.save_button);
+        cancelButton = findViewById(R.id.cancel_save_button);
+
+        /*-----  VM  -----*/
+        mResultVM = ViewModelProviders.of(this).get(ResultVM.class);
+        mResultVM.init();
 
         /*------ LISTENERS ------*/
         cancelButton.setOnClickListener(v -> finish());
         saveButton.setOnClickListener(v -> saveResult());
 
         /*------ INTENT ------*/
         Intent intent = getIntent();
-        position = intent.getIntExtra(WorkoutsListActivity.WORKOUT_ID,1);
-        mWorkout = (Workout) intent.getSerializableExtra(WorkoutsListActivity.WORKOUT_OBJ);
-        Log.i(LOG_TAG, "workout read: "+mWorkout);
+        position = intent.getIntExtra(WorkoutsListActivity.WORKOUT_ID, 1);
+        workout = (Workout) intent.getSerializableExtra(WorkoutsListActivity.WORKOUT_OBJ);
+        Log.i(LOG_TAG, "workout read: " + workout);
 
         // Build a string with exercises
-        StringBuilder exercises = new StringBuilder();
-        ArrayList<String> exercisesArray = mWorkout.getExercises();
-        for (String s : exercisesArray){
-            exercises.append(s).append("\n");
-        }
+        String exercises = WorkoutUtils.buildExerciseString(workout);
 
         // Fill the UI with workout info
-        mName.setText(mWorkout.getTitle());
-        mDescription.setText(mWorkout.getType());
-        mExercises.setText(exercises.toString());
-
+        mName.setText(workout.getTitle());
+        mDescription.setText(workout.getDescription());
+        mExercises.setText(exercises);
         workoutItemSaveButton.setVisibility(GONE);
         workoutItemPlanButton.setVisibility(GONE);
 
-        //TODO finish this activity --> work in progress
+        //change score_type text according to the current workout
+        String scoreType = ResultUtils.setScoreType(workout);
+        mScoreType.setText(scoreType);
+
+        //change edit_score hint according to the current workout
+        String scoreTypeHint = ResultUtils.setScoreTypeHint(workout);
+        mScore.setHint(scoreTypeHint);
+
+        //change edit_score digits according to the current workout
+        String scoreDigits = ResultUtils.setScoreDigits(workout);
+        mScore.setKeyListener(DigitsKeyListener.getInstance(scoreDigits));
```

Comment:
Confused by this line: what does it do? (You can explain to me in the next meeting)

Reply:
or it could be commented a bit better 😝

Reply:
changed
(sorry, thought it was understandable. it's the digits accepted on the input field (idiot-proof feature))